server/supernova/audio_backend/portaudio_backend.hpp

```diff
@@ -124,12 +104,40 @@ class portaudio_backend : public detail::audio_backend_base<sample_type, float,
     bool open_stream(std::string const& input_device, unsigned int inchans, std::string const& output_device,
                      unsigned int outchans, unsigned int samplerate, unsigned int pa_blocksize, int h_blocksize) {
         int input_device_index, output_device_index;
-        if (!match_device(input_device, input_device_index) || !match_device(output_device, output_device_index))
+
```

Comment:
please document this function, at least the behavior of `input_device`, `output_device`, `inchans` and `outchans`, because it's not immediately apparent